fabtests/benchmarks/benchmark_shared.c

```diff
@@ -173,10 +254,22 @@ int run_pingpong(void)
 {
 	int i, ret = 0;
 
+	if (ft_check_opts(FT_OPT_SKIP_MSG_ALLOC) &&
+	    !ft_check_opts(FT_OPT_OOB_SYNC)) {
+		FT_ERR("out-of-band address exchange and synchronization required");
+		return EXIT_FAILURE;
+	}
+
 	ret = ft_init_fabric();
 	if (ret)
 		return ret;
 
+	if (ft_check_opts(FT_OPT_SKIP_MSG_ALLOC)) {
```

Comment:
Yes, please use a separate flag for the tx first behavior.

Reply:
Consuming the extra receive works for me. Will look at changing.

Reply:
Consuming the extra receives worked well. In addition, I was able to support inband sync and exchange.

Reply:
The in-band address exchange working makes sense but I'm confused the in-band sync works as well since that one blocks for the rx before posting one. Can you explain how that works with the flag?

Reply:
Nevermind - I see the extra rx post and comment now